### PR TITLE
Enable generating compile_commands.json

### PR DESCRIPTION
**Pull request by rerobika**
This feature is useful for intellisense users.

Signed-off-by: Robert Fancsik <frobert@inf.u-szeged.hu>

### CODE REVIEW

**Review by akosthekiss — APPROVED** [review]
LGTM

File: CMakeLists.txt [hunk 1]
```diff
@@ -148,6 +148,9 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin/")
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib/")
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib/")
 
+# Enable generating compile_commands.json
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```
Comment (lygstate) [inline CMakeLists.txt:152]: Prefer option instead set.
For building procedure, there is no need generate it

Comment (rerobika) [reply CMakeLists.txt:152]: There is no disadvantage of generating it, and it's useful for some IDEs and intellisense.

Comment (rerobika) [reply CMakeLists.txt:152]: This reasoning would more convince me if the recommended build platform would be the low-end device. However, most of the time the (cross)-build is performed on the host device where this extra json doesn't add to much for the "pollution" what cmake generates.

Comment (akosthekiss) [reply CMakeLists.txt:152]: @lygstate What exactly do you mean by pollution? The file is generated into the build tree which is already full of cmake-generated auxiliary files.

Comment (lygstate) [reply CMakeLists.txt:152]: > @lygstate What exactly do you mean by pollution? The file is generated into the build tree which is already full of cmake-generated auxiliary files.

The parent project also generating compile_commands.json

Comment (akosthekiss) [reply CMakeLists.txt:152]: @lygstate Is that a real issue anywhere?

@rerobika How about finding a middle ground and adding a cli option to build.py that controls this cmake variable?

Comment (rerobika) [reply CMakeLists.txt:152]: I can deal with that if it is enabled by default.

Comment (lygstate) [reply CMakeLists.txt:152]: option(CMAKE_EXPORT_COMPILE_COMMANDS "Enable export compile commands for jerryscript" ON)

Comment (akosthekiss) [reply CMakeLists.txt:152]: @lygstate Thanks, we _do_ know how to add an option to cmake. The suggestion was to add an option to `build.py` (we know how to add that, too). What we do not know whether the "pollution" is a real issue in any embedding project?

**Comment by lygstate** [comment]
LGTM

**Review by dbatyai — APPROVED** [review]
LGTM

**Pull request merged**